src/agent/debuglet.ts

```diff
@@ -322,6 +333,17 @@ export class Debuglet extends EventEmitter {
 
     });
   }
+  checkReady() {
+    const diff = Date.now() - this.promiseResolvedTimestamp;
+    if (diff > PROMISE_RESOLVE_CUT_OFF_IN_MILLISECONDS) {
```

Comment:
I think this should be

```javascript
  if (!this.promise || diff > PROMISE_RESOLVE_CUT_OFF_IN_MILLISECONDS) {
```

Reply:
Understood. but since we set the initial this.promiseResolvedTimestamp to be -Infinite. We can be sure that the first diff will always > PROMISE_RESOLVE_CUT_OFF_IN_MILLISECONDS, right?

Reply:
Makes sense. But since that contract can easily be broken in the future, what about adding an assert at line 345 that `this.promise` has some value?